### ISSUE TITLE
Rule to prefer passing a function instead of a string literal, when equivalent

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [vitest-dev/eslint-plugin-vitest#677]
I often write tests for named functions in a `describe()` block that takes in the function name:

```js
import { myFunction } from "./myFunction.js";

describe("myFunction", () => { /* ... */ });
```

But, `describe()` allows taking in a `Function`. IMO it's better to pass the function itself so that if the function gets renamed, the test automatically uses the updated name 

```js
import { myFunctionRenamed } from "./myFunctionRenamed.js";

describe("myFunction", () => { /* ... */ }); // old name 🙁
describe(myFunctionRenamed, () => { /* ... */ }); // new name ☺️
```

Proposal: how about a lint rule that enforces passing a function rather than a string literal when they're equivalent?

It's a little hard to say exactly what equivalency here means. Maybe if the following cases are met:

* The file name matches a function in root-level scope (e.g. `"myFunction.test.ts"` -> `myFunction`)
* The file name also matches the string literal passed to `describe()`

**Comment by veritem** [comment]
I'll give it a review if you're willing to write it. I don't think it's that hard since we now have setup to enable typechecking for rules.
